### DISCUSSION TITLE
Do I have to use a datapack to show the scoreboard objectives?

### DISCUSSION THREAD

**Discussion by x0tester0x** [Dantevg/WebStats#97] [Q&A] [answered]
Do I have to use a datapack to show the scoreboard objectives?

**Comment by Dantevg** [comment]
If you already have existing scoreboard objectives, you don't need anything else to show them on WebStats. You can just list them in the config file, but by default it should already display all scoreboard objectives it can find.

**Reply by x0tester0x** [reply]
OK, how can I check if I have already existing scoreboard objectives, because I do not use any datapacks and the website does not display anything?

**Reply by Dantevg** [reply]
If you type `/scoreboard objectives setdisplay sidebar ` in the command bar, it will suggest you the list of all objectives that you have.

**Reply by x0tester0x** [reply]
OK, i will test this.
The available scoreboard objectives are these here https://minecraft.fandom.com/wiki/Statistics correct?

**Reply by x0tester0x** [reply]
OK I did run the command `/scoreboard objectives setdisplay sidebar` but it returns `Nothing changed. That display slot is already empty` ...

**Reply by Dantevg** [reply]
Ah no, the statistics are separate from the scoreboard but I get where the confusion is coming from. If you want to display those statistics, you can use PlaceholderAPI and [the statistic placeholder](https://wiki.placeholderapi.com/users/placeholder-list/#statistic). Not sure if you're familiar with PlaceholderAPI, just let me know if you need help with it.


By the way, the Minecraft wiki has moved to https://minecraft.wiki/w/Statistics (it's no longer on fandom) 🙂

**Reply by Dantevg** [reply]
> OK I did run the command `/scoreboard objectives setdisplay sidebar` but it returns `Nothing changed. That display slot is already empty` ...

I meant to only type it in (end with a space) and don't press enter yet, it will display a list of suggestions (if you're in-game).

**Reply by x0tester0x** [reply]
> > OK I did run the command `/scoreboard objectives setdisplay sidebar` but it returns `Nothing changed. That display slot is already empty` ...
> 
> I meant to only type it in (end with a space) and don't press enter yet, it will display a list of suggestions (if you're in-game).

OK, I did test this now but I get only some team.color scoreboard objectives suggestions but nothing else, what do I miss here?

**Reply by Dantevg** [reply]
That would be normal if you haven't set up any scoreboard objectives. If you want to display the statistics you linked, you can completely ignore the scoreboard objectives and set up PlaceholderAPI. Hopefully the comments in the config.yml and the [wiki](https://github.com/Dantevg/WebStats/wiki/Getting-Started#placeholderapi) [pages](https://github.com/Dantevg/WebStats/wiki/Config-file#PlaceholderAPI-connectivity) explain things well enough

**Reply by x0tester0x** [reply]
OK, I understand now, I have to manually setup some scoreboard objectives to display them. I thought there are already some default scoreboard objectives setup as default. So then when I do not have any scoreboard objectives setup also this plugin https://www.spigotmc.org/resources/placeholderapi-scoreboard-objectives-placeholder.48236/ will not display anything, correct?

**Reply by Dantevg** [reply]
Looks like it will also not display anything, indeed.

**Reply by x0tester0x** [reply]
A datapack like the [Vanilla Tweaks](https://vanillatweaks.net/) Track Raw Statistics and Track Statistics will then setup these scoreboard objectives, correct?

**Reply by Dantevg** [reply]
Yes exactly. Though keep in mind that those datapacks will only start tracking the statistics after they are installed, so all scores will begin counting at 0.

**Reply by x0tester0x** [reply]
OK, perfect, so now what is the exact difference between scoreboard objectives and statistics (because both can display the same values)?

**Reply by Dantevg** [reply, marked as answer]
Scoreboard objectives are server-side and can hold any (numeric) value. They can be updated automatically (for some [criteria](https://minecraft.wiki/w/Scoreboard#Criteria)) but also manually by commands (includes datapacks) and plugins. They are indexed by name, which can be a player's name but doesn't need to be (datapacks often use an objective called "constants" which holds the value 100 for the entry with name "100" and so on).

The statistics are client-side and are thus per player. They only store the statistics that are defined in the Minecraft code and as such cannot hold other values and the values cannot be changed by commands.

The scoreboard is meant to store any data that a datapack author or anyone might want to store, while the statistics are specifically storing, well, the player's statistics.

**Reply by x0tester0x** [reply]
> A datapack like the [Vanilla Tweaks](https://vanillatweaks.net/) Track Raw Statistics and Track Statistics will then setup these scoreboard objectives, correct?

Is there somewhere a documentation which scorboard objectives Vanilla Tweaks Track Raw Statistics and Track Statistics will setup?

**Reply by Dantevg** [reply]
There doesn't seem to be a list online (in general, I find Vanilla Tweaks' documentation to be very lacking.

These are the objectives listed in the `load.mcfunction` files in these datapacks (for version 1.1.5 and 1.7.0, not the latest version, just the ones I had lying around):

<details>
<summary>Track Statistics</summary>

Most of these ones also have a "show" variant which you should use, the non-show variants will flicker if displayed in the sidebar for instance. Using the `/scoreboard objectives setdisplay sidebar ` autocomplete list is useful for this.

- Elytra km (`hc_elytraKm`)
- Elytra cm (`hc_elytraCm`)
- Play Minutes (`hc_playTime`)
- Play Ticks (`hc_playTick`)
- Swim km (`hc_swimKm`)
- Swim (`hc_swimCm`)
- Mine Ores (`hc_MineOre`)
- Mine All Coal (`hc_MineCoal`)
- Mine All Diamond (`hc_MineDiamond`)
- Mine All Emerald (`hc_MineEmerald`)
- Mine All Redstone (`hc_MineRedstone`)
- Mine All Lapis (`hc_MineLapis`)
- Mine All Iron (`hc_MineIron`)
- Mine All Gold (`hc_MineGold`)
- Mine All Copper (`hc_MineCopper`)
- Mine Stone Coal (`hc_MineSCoal`)
- Mine Stone Diamond (`hc_MineSDiamond`)
- Mine Stone Emerald (`hc_MineSEmerald`)
- Mine Quartz (`hc_MineQuartz`)
- Mine Stone Redstone (`hc_MineSRedstone`)
- Mine Stone Lapis (`hc_MineSLapis`)
- Mine Stone Iron (`hc_MineSIron`)
- Mine Stone Gold (`hc_MineSGold`)
- Mine Stone Copper (`hc_MineSCopper`)
- Mine Ancient Debris (`hc_MineNetherite`)
- Mine Deepslate Coal (`hc_MineDCoal`)
- Mine Deepslate Diamond (`hc_MineDDiamond`)
- Mine Deepslate Emerald (`hc_MineDEmerald`)
- Mine Deepslate Redstone (`hc_MineDRedstone`)
- Mine Deepslate Lapis (`hc_MineDLapis`)
- Mine Deepslate Iron (`hc_MineDIron`)
- Mine Deepslate Gold (`hc_MineDGold`)
- Mine Deepslate Copper (`hc_MineDCopper`)
</details>

<details>
<summary>Track Raw Statistics</summary>

- Deaths (`ts_Deaths`)
- Kill Count (`ts_KillCount`)
- Total Kills (`ts_TotalKills`)
- Animals Bred (`ts_AnimalsBred`)
- Aviate (`ts_Aviate`)
- Brewing (`ts_Brewing`)
- Cake Eaten (`ts_CakeEaten`)
- Chest Opened (`ts_ChestOpened`)
- Climb (`ts_Climb`)
- Crafts (`ts_Crafts`)
- Crouch (`ts_Crouch`)
- Damage Dealt (`ts_DamageDealt`)
- Damage Taken (`ts_DamageTaken`)
- Ender Chest (`ts_EnderChest`)
- Fall (`ts_Fall`)
- Fish Caught (`ts_FishCaught`)
- Flower Potted (`ts_FlowerPotted`)
- Fly (`ts_Fly`)
- Furnace Used (`ts_FurnaceUsed`)
- Horse Rode (`ts_HorseRode`)
- Items Enchanted (`ts_EnchantItem`)
- Jump (`ts_Jump`)
- Minecart Ride (`ts_MinecartRide`)
- Mob Kills (`ts_MobKills`)
- Play Noteblock (`ts_PlayNoteblock`)
- Tune Noteblock (`ts_TuneNoteblock`)
- Pig Ride (`ts_PigRide`)
- Play Time (`ts_PlayTime`)
- Player Kills (`ts_PlayerKills`)
- Records Played (`ts_RecordsPlayed`)
- Shulker Box (`ts_ShulkerBox`)
- Sneak (`ts_Sneak`)
- Sprint (`ts_Sprint`)
- Swim (`ts_Swim`)
- Time Since Last Death (`ts_LastDeath`)
- Villager Trade (`ts_VillagerTrade`)
- Trap Chest Open (`ts_TrapChestOpen`)
- Walk (`ts_Walk`)
- Craft Beacon (`ts_CraftBeacon`)
- Craft End Crystal (`ts_CraftEndCryst`)
- Craft Conduit (`ts_CraftConduit`)
- Craft Shulker Box (`ts_CraftShulkBox`)
- Use Potion (`ts_UsePotion`)
- Use Totem (`ts_UseTotem`)
- Use Torch (`ts_UseTorch`)
- Use Golden Apple (`ts_UseGoldApple`)
- Use Bonemeal (`ts_UseBonemeal`)
- Buckets Filled (`ts_UseBucket`)
- Water Buckets Emptied (`ts_UseWatrBucket`)
- Lava Buckets Emptied (`ts_UseLavaBucket`)
- Thrown Snowball (`ts_UseSnowball`)
- Thrown Eye Of Ender (`ts_UseEyeOfEnder`)
- Thrown Ender Pearl (`ts_UseEnderPearl`)
- Thrown Trident (`ts_UseTrident`)
- Thrown Bottle o' Enchanting (`ts_UseBottleEnch`)
- Cast Fishing Rod (`ts_UseFishingRod`)
- Drink Honey Bottle (`ts_UseHoneyBottl`)
- Break Diamond Pickaxe (`ts_BreakPick`)
- Break Diamond Axe (`ts_BreakAxe`)
- Break Diamond Shovel (`ts_BreakShovel`)
- Break Diamond Sword (`ts_BreakSword`)
- Break Diamond Hoe (`ts_BreakHoe`)
- Break Netherite Pickaxe (`ts_BreakNPick`)
- Break Netherite Axe (`ts_BreakNAxe`)
- Break Netherite Shovel (`ts_BreakNShovel`)
- Break Netherite Sword (`ts_BreakNSword`)
- Break Netherite Hoe (`ts_BreakNHoe`)
- Break Bow (`ts_BreakBow`)
- Break Shears (`ts_BreakShears`)
- Mine Coal (`ts_MineCoal`)
- Mine Diamond (`ts_MineDiamond`)
- Mine Emerald (`ts_MineEmerald`)
- Mine Quartz (`ts_MineQuartz`)
- Mine Redstone (`ts_MineRedstone`)
- Mine Lapis (`ts_MineLapis`)
- Mine Iron (`ts_MineIron`)
- Mine Gold (`ts_MineGold`)
- Mine Copper (`ts_MineCopper`)
- Mine Ancient Debris (`ts_MineNetherite`)
- Mine Deepslate Coal (`ts_MineDCoal`)
- Mine Deepslate Diamond (`ts_MineDDiamond`)
- Mine Deepslate Emerald (`ts_MineDEmerald`)
- Mine Deepslate Redstone (`ts_MineDRedstone`)
- Mine Deepslate Lapis (`ts_MineDLapis`)
- Mine Deepslate Iron (`ts_MineDIron`)
- Mine Deepslate Gold (`ts_MineDGold`)
- Mine Deepslate Copper (`ts_MineDCopper`)
- Kill Hoglin (`ts_KillHoglin`)
- Kill Piglin (`ts_KillPiglin`)
- Kill Strider (`ts_KillStrider`)
- Kill Zoglin (`ts_KillZoglin`)
- Kill Bee (`ts_KillBee`)
- Kill Bat (`ts_KillBat`)
- Kill Blaze (`ts_KillBlaze`)
- Kill Cave Spider (`ts_KillCveSpider`)
- Kill Chicken (`ts_KillChicken`)
- Kill Cod (`ts_KillCod`)
- Kill Cow (`ts_KillCow`)
- Kill Creeper (`ts_KillCreeper`)
- Kill Dolphin (`ts_KillDolphin`)
- Kill Donkey (`ts_KillDonkey`)
- Kill Drowned (`ts_KillDrowned`)
- Kill Elder Guardian (`ts_KillEGuardian`)
- Kill Ender Dragon (`ts_KillEDragon`)
- Kill Enderman (`ts_KillEnderman`)
- Kill Endermite (`ts_KillEndermite`)
- Kill Evoker (`ts_KillEvoker`)
- Kill Ghast (`ts_KillGhast`)
- Kill Guardian (`ts_KillGuardian`)
- Kill Horse (`ts_KillHorse`)
- Kill Husk (`ts_KillHusk`)
- Kill Illusioner (`ts_KillIllusion`)
- Kill Iron Golem (`ts_KillIronGolem`)
- Kill Llama (`ts_KillLlama`)
- Kill Magma Cube (`ts_KillMagmaCube`)
- Kill Mooshroom (`ts_KillMooshroom`)
- Kill Mule (`ts_KillMule`)
- Kill Ocelot (`ts_KillOcelot`)
- Kill Parrot (`ts_KillParrot`)
- Kill Phantom (`ts_KillPhantom`)
- Kill Pig (`ts_KillPig`)
- Kill Zombified Piglin (`ts_KillPigman`)
- Kill Polar Bear (`ts_KillPolarBear`)
- Kill Pufferfish (`ts_KillPuffish`)
- Kill Rabbit (`ts_KillRabbit`)
- Kill Salmon (`ts_KillSalmon`)
- Kill Sheep (`ts_KillSheep`)
- Kill Shulker (`ts_KillShulker`)
- Kill Silverfish (`ts_KillSilvfish`)
- Kill Skeleton (`ts_KillSkeleton`)
- Kill Skeleton Horse (`ts_KillSkeletonH`)
- Kill Slime (`ts_KillSlime`)
- Kill Snow Golem (`ts_KillSnowGolem`)
- Kill Spider (`ts_KillSpider`)
- Kill Squid (`ts_KillSquid`)
- Kill Stray (`ts_KillStray`)
- Kill Tropical Fish (`ts_KillTropifish`)
- Kill Turtle (`ts_KillTurtle`)
- Kill Vex (`ts_KillVex`)
- Kill Villager (`ts_KillVillager`)
- Kill Vindicator (`ts_KillVindicatr`)
- Kill Witch (`ts_KillWitch`)
- Kill Wither (`ts_KillWither`)
- Kill Wither Skeleton (`ts_KillWSkeleton`)
- Kill Wolf (`ts_KillWolf`)
- Kill Zombie (`ts_KillZombie`)
- Kill Zombie Horse (`ts_KillZombieH`)
- Kill Zombie Villager (`ts_KillZombieV`)
- Kill Wandering Trader (`ts_KillWandering`)
- Kill Trader Llama (`ts_KillTraderLla`)
- Kill Fox (`ts_KillFox`)
- Kill Panda (`ts_KillPanda`)
- Kill Pillager (`ts_KillPillager`)
- Kill Ravager (`ts_KillRavager`)
- Kill Glow Squid (`ts_KillGlowSquid`)
- Kill Axolotl (`ts_KillAxolotl`)
- Kill Goat (`ts_KillGoat`)
- Kill Warden (`ts_KillWarden`)
- Kill Camel (`ts_KillCamel`)
- Kill Sniffer (`ts_KillSniffer`)
- Killed by Hoglin (`ts_DthHoglin`)
- Killed by Piglin (`ts_DthPiglin`)
- Killed by Strider (`ts_DthStrider`)
- Killed by Zoglin (`ts_DthZoglin`)
- Killed by Bee (`ts_DthBee`)
- Killed by Blaze (`ts_DthBlaze`)
- Killed by Cave Spider (`ts_DthCveSpider`)
- Killed by Creeper (`ts_DthCreeper`)
- Killed by Dolphin (`ts_DthDolphin`)
- Killed by Drowned (`ts_DthDrowned`)
- Killed by Elder Guardian (`ts_DthEGuardian`)
- Killed by Ender Dragon (`ts_DthEDragon`)
- Killed by Enderman (`ts_DthEnderman`)
- Killed by Endermite (`ts_DthEndermite`)
- Killed by Evoker (`ts_DthEvoker`)
- Killed by Ghast (`ts_DthGhast`)
- Killed by Guardian (`ts_DthGuardian`)
- Killed by Husk (`ts_DthHusk`)
- Killed by Illusioner (`ts_DthIllusion`)
- Killed by Iron Golem (`ts_DthIronGolem`)
- Killed by Llama (`ts_DthLlama`)
- Killed by Magma Cube (`ts_DthMagmaCube`)
- Killed by Phantom (`ts_DthPhantom`)
- Killed by Zombified Piglin (`ts_DthPigman`)
- Killed by Polar Bear (`ts_DthPolarBear`)
- Killed by Pufferfish (`ts_DthPuffish`)
- Killed by Rabbit (`ts_DthRabbit`)
- Killed by Shulker (`ts_DthShulker`)
- Killed by Silverfish (`ts_DthSilvfish`)
- Killed by Skeleton (`ts_DthSkeleton`)
- Killed by Skeleton Horse (`ts_DthSkeletonH`)
- Killed by Slime (`ts_DthSlime`)
- Killed by Snow Golem (`ts_DthSnowGolem`)
- Killed by Spider (`ts_DthSpider`)
- Killed by Stray (`ts_DthStray`)
- Killed by Turtle (`ts_DthTurtle`)
- Killed by Vex (`ts_DthVex`)
- Killed by Villager (`ts_DthVillager`)
- Killed by Vindicator (`ts_DthVindicatr`)
- Killed by Witch (`ts_DthWitch`)
- Killed by Wither (`ts_DthWither`)
- Killed by Wither Skeleton (`ts_DthWSkeleton`)
- Killed by Wolf (`ts_DthWolf`)
- Killed by Zombie (`ts_DthZombie`)
- Killed by Zombie Horse (`ts_DthZombieH`)
- Killed by Zombie Villager (`ts_DthZombieV`)
- Killed by Wandering Trader (`ts_DthWandering`)
- Killed by Trader Llama (`ts_DthTraderLla`)
- Killed by Fox (`ts_DthFox`)
- Killed by Panda (`ts_DthPanda`)
- Killed by Pillager (`ts_DthPillager`)
- Killed by Ravager (`ts_DthRavager`)
- Killed by Warden (`ts_DthWarden`)
</details>

**Reply by x0tester0x** [reply]
> That would be normal if you haven't set up any scoreboard objectives. If you want to display the statistics you linked, you can completely ignore the scoreboard objectives and set up PlaceholderAPI. Hopefully the comments in the config.yml and the [wiki](https://github.com/Dantevg/WebStats/wiki/Getting-Started#placeholderapi) [pages](https://github.com/Dantevg/WebStats/wiki/Config-file#PlaceholderAPI-connectivity) explain things well enough

OK so I do only have to add the PlaceholderAPI values and then it should work or do I also have to change the table columns (names) and the column units?

**Reply by Dantevg** [reply]
You only need to add the placeholders to [this list](https://github.com/Dantevg/WebStats/blob/master/src/main/resources/config.yml#L98) (remember to uncomment). The `tables` config is completely optional and defaults to showing everything in a single table. Also the column units are optional, by default all values are unit-less plain values.
